spec.emu

```diff
@@ -136,189 +175,307 @@ location: https://github.com/tc39/proposal-decimal/
     <emu-clause id="sec-decimal.prototype.subtract">
       <h1>Decimal128.prototype.subtract ( _x_ )</h1>
-      <p>Subtracts *_x_* from *this* value according to the rules specified in the IEEE 754 Decimal128 specification. Neither *this* value nor *_x_* is modified. If, in the computation, more significant digits are generated than can be stored in an IEEE 754 Decimal128 value, the final digit will be computed by rounding according to <emu-xref href="#dfn-decimal128-default-rounding-mode">the default rounding mode</emu-xref>.</p>
+      <p>This method performs the following steps when called:</p>
+      <emu-alg>
+        1. Let _O_ be the **this** value.
+        1. Perform ? RequireInternalSlot(_O_, [[Decimal128Data]]).
+        1. Perform ? RequireInternalSlot(_x_, [[Decimal128Data]]).
+        1. Let _d1_ be _O_.[[Decimal128Data]].
+        1. Let _d2_ be _x_.[[Decimal128Data]].
+        1. If _d1_ or _d2_ is *NaN*<sub>D</sub>, return Decimal128ValueToObject(*NaN*<sub>D</sub>).
+        1. If _d1_ is *+∞*<sub>D</sub>, then
+          1. If _d2_ is *+∞*<sub>D</sub>, return Decimal128ValueToObject(*+∞*<sub>D</sub>).
+          1. If _d2_ is *-∞*<sub>D</sub>, return Decimal128ValueToObject(*NaN*<sub>D</sub>).
+          1. Return Decimal128ValueToObject(*+∞*<sub>D</sub>).
+        1. If _d1_ is *-∞*<sub>D</sub>, then
+          1. If _d2_ is *+∞*<sub>D</sub>, return Decimal128ValueToObject(*NaN*<sub>D</sub>).
+          1. If _d2_ is *-∞*<sub>D</sub>, return Decimal128ValueToObject(*-∞*<sub>D</sub>).
+          1. Return Decimal128ValueToObject(*-∞*<sub>D</sub>).
+        1. Return Decimal128ValueToObject(« d1[0] - d2[0], min(d1[1], d2[1]) »).
+      </emu-alg>
     </emu-clause>
     <emu-clause id="sec-decimal.prototype.multiply">
       <h1>Decimal128.prototype.multiply ( _x_ )</h1>
-      <p>Multiplies *this* value by *_x_* according to the rules specified in the IEEE 754 Decimal128 specification. Neither argument is modified. If, in the computation, more significant digits are generated than can be stored in an IEEE 754 Decimal128 value, the final digit will be computed by rounding according to <emu-xref href="#dfn-decimal128-default-rounding-mode">the default rounding mode</emu-xref>.</p>
+      <p>This method performs the following steps when called:</p>
+      <emu-alg>
+        1. Let _O_ be the **this** value.
+        1. Perform ? RequireInternalSlot(_O_, [[Decimal128Data]]).
+        1. Perform ? RequireInternalSlot(_x_, [[Decimal128Data]]).
+        1. Let _d1_ be _O_.[[Decimal128Data]].
+        1. Let _d2_ be _x_.[[Decimal128Data]].
+        1. If _d1_ or _d2_ is *NaN*<sub>D</sub>, return Decimal128ValueToObject(*NaN*<sub>D</sub>).
+        1. If _d1_ is *+∞*<sub>D</sub>, then
+          1. If _d2_ is *+∞*<sub>D</sub>, return Decimal128ValueToObject(*+∞*<sub>D</sub>).
+          1. If _d2_ is *-∞*<sub>D</sub>, return Decimal128ValueToObject(*NaN*<sub>D</sub>).
+          1. Return Decimal128ValueToObject(*+∞*<sub>D</sub>).
```

Comment:
Does this need to flip the sign if _d2_ < 0? How about 0 * ±∞?

Reply:
Right, we do need to flip the sign. 0 * +/-∞ should be NaN. Good spot.